### PR TITLE
support comparison of commonform data types directly using operators

### PR DESCRIPTION
**Pull request by sanket0896**
- override existing operators to use custom operators
- add support to directly compare currency, duration, date, phone-number, array using relational operators
- added a lot of test cases around it and verified that all of them are passing

### CODE REVIEW

File: sd-custom/custom-operator/equals.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  }
+
+  if (lType !== SD_CUSTOM_TYPES.phoneNumber) {
+    return false;
+  }
+
+  return (
+    l.number === r.number &&
+    l.code === r.code &&
+    l.country_code === r.country_code
```
Comment (anirudhvarma12) [inline sd-custom/custom-operator/equals.js:64]: Do we always have both code and country_code? What's the difference b/w the 2?

Comment (sanket0896) [reply sd-custom/custom-operator/equals.js:64]: I took this from `isPhoneNumberEqual` in `apps/spotdraft-fe/src/app/core/utils/blank-json-compare.ts`.

I think we can safely remove the `country_code` check. From what I could see in draftkit, `country_code` is only used to render the flags in the phone number input component.

Comment (sanket0896) [reply sd-custom/custom-operator/equals.js:64]: Can you please confirm whether I should remove it @anirudhvarma12

Comment (anirudhvarma12) [reply sd-custom/custom-operator/equals.js:64]: Yeah, lets do that.

Comment (sanket0896) [reply sd-custom/custom-operator/equals.js]: Made the change. Please squash and merge if everything looks fine.

**Pull request merged**